src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -506,14 +520,51 @@ where
             .make_channel(Duration::from_nanos(self.config.connect_timeout))
             .await?;
 
-        let mut client = BigQueryWriteClient::with_interceptor(
+        let client = BigQueryWriteClient::with_interceptor(
             channel,
             AuthInterceptor {
                 token_provider: T::default(),
             },
         );
+        self.client = Some(client);
+
+        Ok(true)
+    }
+
+    fn auto_ack(&self) -> bool {
+        false
+    }
+}
 
-        let write_stream = client
+impl<
+        T: TokenProvider + 'static,
+        TChannel: tonic::codegen::Service<
+                http::Request<tonic::body::BoxBody>,
+                Response = http::Response<tonic::transport::Body>,
+                Error = TChannelError,
+            > + Send
+            + Clone
+            + 'static,
+        TChannelError: Into<Box<dyn std::error::Error + Send + Sync + 'static>> + Send + Sync,
+    > GbqSink<T, TChannel>
+where
+    TChannel::Future: Send,
+{
```

Comment:
Yep, I'm in the process of thinking about how to do it right now. This isn't as easy as it seems because in the naïve implementation we'd need both a read-only reference to self (for `self.write_streams`) and a mutable one (for `self.get_or_create_write_stream(...)`), which leads to unhappy borrow checker. Ideas are welcome ;)

Reply:
ja that's where the entry api comes into play: https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with

Reply:
but that's an issue, because `get_or_create_write_stream` is async, so we'd need to run this as blocking, which I'm not sure is a good idea 🤔 . Unless there's an async version of `or_insert_with` that I'm not aware of?

Reply:
that can be worked around: https://users.rust-lang.org/t/async-or-insert-with/73320